pkg/cloud/metadata_k8s.go

```diff
@@ -45,9 +45,9 @@ func KubernetesAPIInstanceInfo(clientset kubernetes.Interface) (*Metadata, error
 		return nil, fmt.Errorf("node providerID empty, cannot parse")
 	}
 
-	awsRegionRegex := "([a-z]{2}(-gov)?)-(central|(north|south)?(east|west)?)-[0-9]"
-	awsAvailabilityZoneRegex := "([a-z]{2}(-gov)?)-(central|(north|south)?(east|west)?)-[0-9][a-z]"
-	awsInstanceIDRegex := "i-[a-z0-9]+$"
+	awsRegionRegex := "(snow)|(([a-z]{2}(-gov)?)-(central|(north|south)?(east|west)?)-[0-9])"
+	awsAvailabilityZoneRegex := "(snow)|(([a-z]{2}(-gov)?)-(central|(north|south)?(east|west)?)-[0-9][a-z])"
+	awsInstanceIDRegex := "s.i-[a-z0-9]+|i-[a-z0-9]+$"
```

Comment:
is the "." here literal? if so we need to escape this like so: `awsInstanceIDRegex := "s\.i-[a-z0-9]+|i-[a-z0-9]+$"`

Reply:
Ack, made changes.